type-generator/src/main/java/io/mesosphere/types/dtr/Main.java

```diff
@@ -0,0 +1,156 @@
+package io.mesosphere.types.dtr;
+
+import io.mesosphere.types.dtr.compiler.impl.StringTemplateEngine;
+import io.mesosphere.types.dtr.compiler.internal.CompiledFragment;
+import io.mesosphere.types.dtr.compiler.internal.CompilerEngine;
+import io.mesosphere.types.dtr.compiler.internal.CompiledFragmentType;
+import io.mesosphere.types.dtr.compiler.internal.arguments.FileObjectTypeFragmentArguments;
+import io.mesosphere.types.dtr.models.internal.Type;
+import io.mesosphere.types.dtr.models.internal.TypeScope;
+import io.mesosphere.types.dtr.models.internal.TypeScopeRelation;
+import io.mesosphere.types.dtr.models.internal.scopes.FileScope;
+import io.mesosphere.types.dtr.models.internal.scopes.GlobalScope;
+import io.mesosphere.types.dtr.models.internal.types.*;
+import io.mesosphere.types.dtr.models.parser.RAMLParser;
+import io.mesosphere.types.dtr.repository.internal.Repository;
+import io.mesosphere.types.dtr.repository.internal.RepositoryView;
+import org.apache.commons.cli.*;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+
+    /**
+     * Entry point
+     * @param args
+     */
+    public static void main(String[] args) throws ParseException {
+        Options options = new Options();
+        CommandLineParser parser = new DefaultParser();
+
+        // Prepare options
+        Option o_help = Option.builder("h")
+                .longOpt("help")
+                .desc("Print this help message")
+                .argName("help")
+                .build();
+
+        Option o_project = Option.builder("p")
+                .longOpt("project")
+                .required()
+                .desc("The project name to target")
```

Comment:
What does this mean? Is this a target folder?

Reply:
This comes from the type registry repository idea, in order to keep things versioned (this assumes that the types reside outside from marathon). The path components are:

`<repository dir>/<project>/<version>/{types|compiler|migrations}`

* `-r` defines the repository as a URI (ex. file:/path/to/repository) for LocalRepository
* `-p` defines the project (pointless in our case) --> `marathon`
* `-v` defines the version of the API to generate (ex. `1`). It's just the folder name
* `/types` : Contains the RAML types
* `/compiler` : Contains the compiler templates
* `/migrations` : (will) contain the migration rules between API versions

Reply:
Hm, ok, this should be documented somewhere. However, I'd drop the versioning for now. That seems quite out of scope.